unit-tests/live/rec-play/playback_helper.py

```diff
@@ -0,0 +1,53 @@
+# License: Apache 2.0. See LICENSE file in root directory.
+# Copyright(c) 2021 Intel Corporation. All Rights Reserved.
+
+
+import time
+from rspy import log, test
+from rspy.timer import Timer
+
+'''
+This class is used to wait and verify a requested playback status arrived whithin a requested timeout.
+It blocked the calling thread and sample the playback status using a playback status callback from LRS API\
+
+Note: LRS  set_status_changed_callback function is a 'register' and not a 'set' function
+      meaning multiple set called will add a callbak to the callback list
+      
+Usage should look like this:
+    psv = PlaybackStatusVerifier( device );
+    psv.wait_for_status( 5, rs.playback_status.playing )
+
+'''
+class PlaybackStatusVerifier:
+
+    def __init__( self, dev ):
+        self._current_status = None
+        self._status_changes_cnt = 0
+        playback_dev = dev.as_playback()
+        playback_dev.set_status_changed_callback( self.__signal_on_status_change )
+
+    def __signal_on_status_change( self, playback_status ):
+        log.d('playback status callback invoked with', playback_status)
+        self._status_changes_cnt += 1
+        self._current_status = playback_status
+
+    def wait_for_status( self, timeout, required_status ):
+        status_changes_cnt = self._status_changes_cnt
+        self._current_status = None
+        required_status_detected = False
+        wait_for_event_timer = Timer(timeout)
+        wait_for_event_timer.start()
+        log.d('timeout set to', timeout, '[sec]')
+        while not wait_for_event_timer.has_expired():
+            if required_status == self._current_status:
+                log.d('Required status "' + str(required_status) + '" detected!')
+                required_status_detected = True
+                break
+            # sleep only 100 [ms],  We don't want to miss a status change if more than 1 status change occurs whithin the sleep time
+            time.sleep( 0.1 )
+
+        #If the status changes too fast let say , Stopped -> Playing --> Stopped ,
+        # we want fail the test as our sleep is too long and we will not catch the required status
+        test.check( status_changes_cnt + 1 == self._status_changes_cnt,
```

Comment:
Not sure I understand...
Stopped -> Playing -> Paused -> Playing
If you're waiting for Playing, more than one count is not necessarily bad... am I misunderstanding?

Reply:
The issue is with Stopped -> Playing -> Stopped (because we sample it after a sleep)
On this case we can miss the Playing status.

It's more tailored for a single status check and not for a whole process.
(an expected transition check)

I can find a way to expend it for more usages if you think it's necessary, for the tests in the current usage I expect 1 change.

Reply:
I think the logic behind a check that can make a test fail needs to be good.
We can talk about it if you want.
Up to you.

Reply:
If you do choose to keep it, please note the limitation at the very least

Reply:
I change it to use a 10 [ms] sample rate (can be overridden by the user) and add comments about limitations and how to use this function.
This logic was chosen because LRS use register callback instead of set, otherwise I would have accepted the required status at the Ctor and verify its arrival at the callback.
This logic is good enough for our test considering this behavior.
1000/1000 iteration passed on a local test.